packages/google_maps_flutter/google_maps_flutter_ios/example/shared/maps_example_dart/lib/ground_overlay.dart

```diff
@@ -0,0 +1,333 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// ignore_for_file: public_member_api_docs
+
+import 'package:flutter/material.dart';
+import 'package:google_maps_flutter_platform_interface/google_maps_flutter_platform_interface.dart';
+
+import 'example_google_map.dart';
+import 'page.dart';
+
+enum _GroundOverlayPlacing { position, bounds }
+
+class GroundOverlayPage extends GoogleMapExampleAppPage {
+  const GroundOverlayPage({Key? key})
+      : super(const Icon(Icons.map), 'Ground overlay', key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return const GroundOverlayBody();
+  }
+}
+
+class GroundOverlayBody extends StatefulWidget {
+  const GroundOverlayBody({super.key});
+
+  @override
+  State<StatefulWidget> createState() => GroundOverlayBodyState();
+}
+
+class GroundOverlayBodyState extends State<GroundOverlayBody> {
+  GroundOverlayBodyState();
+
+  ExampleGoogleMapController? controller;
+  GroundOverlay? _groundOverlay;
+
+  final LatLng _mapCenter = const LatLng(37.422026, -122.085329);
+
+  _GroundOverlayPlacing _placingType = _GroundOverlayPlacing.bounds;
+
+  // Positions for demonstranting placing ground overlays with position, and
+  // changing positions.
+  final LatLng _groundOverlayPos1 = const LatLng(37.422026, -122.085329);
+  final LatLng _groundOverlayPos2 = const LatLng(37.42, -122.08);
+  late LatLng _currentGroundOverlayPos;
+
+  // Bounds for demonstranting placing ground overlays with bounds, and
+  // changing bounds.
+  final LatLngBounds _groundOverlayBounds1 = LatLngBounds(
+      southwest: const LatLng(37.42, -122.09),
+      northeast: const LatLng(37.423, -122.084));
+  final LatLngBounds _groundOverlayBounds2 = LatLngBounds(
+      southwest: const LatLng(37.421, -122.091),
+      northeast: const LatLng(37.424, -122.08));
+  late LatLngBounds _currentGroundOverlayBounds;
+
+  Offset _anchor = const Offset(0.5, 0.5);
+
+  Offset _dimensions = const Offset(1000, 1000);
+
+  // Index to be used as identifier for the ground overlay.
+  // If position is changed to bounds and vice versa, the ground overlay will
+  // be removed and added again with the new type. Also anchor can be given only
+  // when the ground overlay is created with position and cannot be change after
```

Comment:
Typo (might need to check line length after changing it):
```suggestion
  // when the ground overlay is created with position and cannot be changed after
```